core/src/main/java/org/mule/runtime/core/api/rx/Exceptions.java

```diff
@@ -158,6 +158,19 @@ public static MuleException rxExceptionToMuleException(Throwable throwable)
   }
 
   /**
+   * Unwrap a particular {@code Throwable}.
+   * 
+   * @param throwable the exception to wrap
+   * @return the unwrapped exception
+   */
+  public static Throwable unwrap(Throwable throwable) {
+    while (throwable.getClass().getName().equals(REACTIVE_EXCEPTION_CLASS_NAME)) {
```

Comment:
Before we used an implementation of unwrap exposed by reactor, but there are class-loader issues which means that sometimes `instanceof` fails and the exception does not get unwrapped. I discussed this with @pablokraan